### DISCUSSION TITLE
CSV importer question

### DISCUSSION THREAD

**Discussion by G4Zz0L1** [firefly-iii/firefly-iii#4477] [Firefly III]
Hi, I am running Firefly III version 5.5.0-beta.1 (installed on my vps straight from github repo).
Everything's working fine, but I've some problems with the api auth on the csv-importer.
Is there any known problem related to a folder installation with the importer auth?
I can't create any subdomain, so I'm limited to a folder system (which usually works just fine with everything).
Afaik it seems related to laravel/passport, but I'm not sure.
I've seen in the related table that the expire date it's the same as the creation and it has no scope, but I'm no laravel expert so I don't know if it's related.
I don't have any other clue to what the cause may be, so I'm asking here.

**Extra info**

![image](https://user-images.githubusercontent.com/13220636/110225359-509cdc80-7ee5-11eb-93fc-c642513effad.png)

![image](https://user-images.githubusercontent.com/13220636/110225707-5942e200-7ee8-11eb-8266-daebe2c414ae.png)

Debug information generated at 2021-03-07 01:51:43 Europe/Rome for Firefly III version **5.5.0-beta.1**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.5.0-beta.1 |
| Firefly III API | 1.5.0 |
| PHP | 8.0.3 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | false |
| Telemetry | true |
| Layout | v1 |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| Default language | it_IT |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| DB version | 15 (exp. 15) |
| Login provider | eloquent  |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| Session start | 2021-03-01 00:00:00 |
| Session end | 2021-03-31 23:59:59 |
| Session first | 2021-01-01 00:00:00 |
| User ID | 1 |
| User language | it_IT |
| User locale | it_IT |
| Attempt at "it_IT.utf8" | 'it_IT.utf8' |
| Attempt at "it_IT.UTF-8" | 'it_IT.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.11 Safari/537.36 |

Logs from csv-importer

```
[2021-03-07 01:45:04] local.DEBUG: Now at App\Http\Controllers\IndexController::reset  
[2021-03-07 01:45:05] local.DEBUG: Now at App\Http\Controllers\IndexController::index  
[2021-03-07 01:45:05] local.DEBUG: Base URL   : ""  
[2021-03-07 01:45:05] local.DEBUG: Vanity URL : "https://mydomain.info/firefly"  
[2021-03-07 01:45:05] local.DEBUG: No access token cookie, redirect to token.index  
[2021-03-07 01:45:05] local.DEBUG: Now at App\Http\Controllers\TokenController::index  
[2021-03-07 01:45:05] local.INFO: The following configuration information was found:  
[2021-03-07 01:45:05] local.INFO: Personal Access Token: "eyJ0eXAiOiJKV1QiLCJhbGciO" (limited to 25 chars if present)  
[2021-03-07 01:45:05] local.INFO: Client ID            : "3"  
[2021-03-07 01:45:05] local.INFO: Base URL             : "https://mydomain.info/firefly"  
[2021-03-07 01:45:05] local.INFO: Vanity URL           : "https://mydomain.info/firefly"  
[2021-03-07 01:45:05] local.DEBUG: Found personal access token + URL "https://mydomain.info/firefly" in config, set cookie and return to index.  
[2021-03-07 01:45:06] local.DEBUG: Now at App\Http\Controllers\IndexController::index  
[2021-03-07 01:45:06] local.DEBUG: Base URL   : "https://mydomain.info/firefly"  
[2021-03-07 01:45:06] local.DEBUG: Vanity URL : "https://mydomain.info/firefly"  
[2021-03-07 01:45:06] local.DEBUG: Has access token cookie.  
[2021-03-07 01:45:07] local.DEBUG: Now at App\Http\Controllers\TokenController::doValidate  
```

**Bonus points**

- :heavy_check_mark: I have read the FAQ at https://bit.ly/FF3-FAQ
- :heavy_check_mark: I added a screenshot
- :heavy_check_mark: I added log files 
- :x: I was able to replicate the issue on the demo site.

**Comment by JC5** [comment]
Are you sure you're using URL and not "URI" in your environment variables?

**Reply by G4Zz0L1** [reply]
I don't know what do you mean by that.
If you're talking about the `FIREFLY_III_URL` and `VANITY_URL` env variable of my csv-importer installation, the're configured correctly (as you can see on the logs) with `https://mydomain.info/firefly`.

**Reply by G4Zz0L1** [reply]
Seeing your autosave script (https://github.com/JC5/autosave/blob/main/autosave.php#L327), I've also tried to test if it works with Insomnia, with the same result. I don't know what can be wrong.

![image](https://user-images.githubusercontent.com/13220636/110253817-95c81980-7f8c-11eb-9eb4-39ff91967cb8.png)

**Reply by G4Zz0L1** [reply]
@JC5 any idea on this?

**Reply by JC5** [reply]
I see you're using a sub directory. It shouldn't really matter though. I'm sorry, assuming the token is correct this should work. :(

**Reply by G4Zz0L1** [reply]
Well, the token is returned from a firefly modal when I create a pat (as shown on your guide on https://docs.firefly-iii.org/csv/install/configure/), so I think I've got it right.
I've also tried to create several ones, but it doesn't seems to matter.
I don't think the subfolder is a problem, or I think Insomnia should have said something about it different than "AuthenticationException".
Can you give some direction to debug this?
What else can I try to figure out where the problem is?

**Reply by JC5** [reply]
Mmmm. The challenge is that the authorisation routine is entirely handled by a library, there's literally zero code on my end handling this. All I can think of is that somehow the Authorization header gets stripped or edited. What's Insomnia, btw?

**Reply by G4Zz0L1** [reply]
Insomnia is an api client, like Postman, but open source.
Turns out (after hours of debugging) that was an Apache problem, not related to laravel/passport as I thought.
For a misconfiguration in this alias on Apache, the following rows in the .htaccess on the public folder were causing troubles.
```
RewriteCond %{HTTP:Authorization} .
RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
```
